heartbeat/Filesystem

```diff
@@ -389,6 +389,82 @@ fstype_supported()
 	return $OCF_ERR_INSTALLED
 }
 
+#
+# Determine backend filesystem support state
+#
+backend_support()
+{
+	# network backended service type need retry timeout (default: 15)
+	# please note the timeout value here mean "15 times", not "15 secs"
+	local timeout
+	timeout=${OCF_RESKEY_CRM_meta_timeout:-15000}
+	timeout=$((timeout/1000))
+
+	# 0: success 1: failed (default)
+	local pass=1
+
+	while [ $timeout -gt 0 ]; do
+		case "$FSTYPE" in
+			glusterfs)
+				# it should failed if service not ready
+				# here we set a general timeout value for waiting response
+				# since gluster will have slow response when starting up
+				timeout 5 gluster volume status >/dev/null 2>&1 && pass=0
```

Comment:
error message from GlusterFS is not always helpful, setting a timed out value here is to avoid waiting too long